src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,199 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> addedNode = new Node<T>(value);
+        if (size > 0) {
+            addedNode.prev = tail;
+            tail.next = addedNode;
+            tail = addedNode;
+        }
+        ifEmptyList(addedNode);
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexSizeMore(index);
+        if (index == size) {
+            add(value);
+        } else {
+            addIfListFilled(value, index);
+        }
```

Comment:
You don't need checkIndexSizeMore method
```suggestion
        if (index == size) {
            add(value);
            return;
        }
        checkIndex(index);
        addByIndex(value, index);
```